test/apps/iodemo/io_demo.cc

```diff
@@ -99,8 +140,13 @@ class P2pDemoCommon : public UcxContext {
     /* Asynchronous IO message */
     class IoMessage : public UcxCallback {
     public:
-        IoMessage(size_t buffer_size, io_op_t op, uint32_t sn, size_t data_size) :
-            _buffer(malloc(buffer_size)) {
+        IoMessage() {
+        }
+        
+        void init(size_t buffer_size, io_op_t op, uint32_t sn, size_t data_size) {
+            if (!_buffer){
```

Comment:
> btw, I think it is not needed anymore, since we have a memory pool
We use pool for IoMessage objects, then we don't need to freequently malloc/free IoMessage objects. But to create each IoMessage object, we still need to malloc a buffer for it. We don't use pool method for this since it won't be called frequently. We don't need to malloc the buffer again in the IoMessage object from the second time to use it since the buffer is already malloced.

Reply:
> `buffer == NULL` instead

Fixed. Thanks!